pkg/scheduler/queue/tenant_queues.go

```diff
@@ -155,10 +160,27 @@ func (qb *queueBroker) enqueueRequestFront(request *tenantRequest, tenantMaxQuer
 		return err
 	}
 
-	queuePath := QueuePath{string(request.tenantID)}
+	queuePath, err := makeQueuePath(request)
+	if err != nil {
+		return err
+	}
 	return qb.tenantQueuesTree.EnqueueFrontByPath(queuePath, request)
 }
 
+func makeQueuePath(request *tenantRequest) (QueuePath, error) {
+	if schedulerRequest, ok := request.req.(*SchedulerRequest); ok {
+		if len(schedulerRequest.AdditionalQueueDimensions) > 1 {
+			return nil, ErrTooManyQueueDimensions
+		}
```

Comment:
it doesn't matter but for now it would indicate a bug. If we needed to expand it in the future, we could always just bump this up first before putting the changes into the frontend.

We could also just throw away the extra dimensions (maybe log it so it's not completely silent), but that goes back to strictness. I think for now we'd want to know (by receiving an error) if we were getting extra dimensions somehow.

Reply:
I think this is unnecessary, and will lead to pain later - under what circumstances could we unintentionally introduce extra dimensions?

Alternatively, could we make the limit configurable, so that changing this in the future is less difficult?

Reply:
removed limit altogether.